### ISSUE TITLE
~5 seconds delay in opening tabs received from other devices

### ISSUE THREAD

**Issue by csuciu** [mozilla-mobile/firefox-ios#12678]
## Steps to reproduce
1. Send a tab from desktop to iOS.
2. On iOS, while Firefox is backgrounded tap on the push notification.
### Expected behavior
The app is foregrounded and immediately the tab received from desktop is open. 
### Actual behavior
The app is foregrounded but it takes ~5 seconds before the tab is open.
### Device & build information
* Device: iPhone 12 Pro (16.1.1)
* Build version: 108.0 (23855)
### Notes This is not reproducible on v107.3.
Attachments: 
https://user-images.githubusercontent.com/1866605/207163358-a5a105b1-d80f-484b-af47-7f9cefd77ea9.mp4

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-5424)

**Comment by lmarceau** [comment]
There's indeed a five seconds delay in the [code for this](https://github.com/mozilla-mobile/firefox-ios/blob/fe431f5f5fbec80cddba3a5de0764e6dbd0283de/Client/Frontend/Browser/BrowserViewController.swift#L415-L417). We can lower the delay, but not sure why there was one in the first place.

**Comment by tarikeshaq** [comment]
Hmmm @lmarceau do we know if https://github.com/mozilla-mobile/firefox-ios/pull/12544 made it to 108? It doesn't change this behaviour but I want to make sure I know if this is indeed a regression with that change or not

**Comment by lmarceau** [comment]
@tarikeshaq that was merged in 109, shouldn't relate. I think it's more the code I linked there.

**Comment by nishant2718** [comment]
Just wanted to clarify here. This isn't a regression, and I wouldn't say it's a bug either. 

Previously, the functionality of opening queued tabs was in AppDelegate, also under a five second delay, https://github.com/mozilla-mobile/firefox-ios/pull/12633. We need a delay because tabs that are queued are added to Profile's `TabQueue`. So, we have to wait for profile to finish initializing before we can load queued tabs.

@dnarcese Would you like to have us look into decreasing the delay? if so, I think we can create a separate task for it.

**Comment by lmarceau** [comment]
@dnarcese the 5 seconds delay was there before too yes. The code was moved recently, but it was in both places inside a 5 seconds delay. So technically it should be the same behavior as before from our perspective. As Nish said, we can probably lower the delay if needed.
